dev/com.ibm.ws.logging/src/com/ibm/ws/logging/internal/impl/BaseTraceService.java

```diff
@@ -1205,6 +1205,217 @@ public TeePrintStream(TrOutputStream trStream, boolean autoFlush) {
             super(trStream, autoFlush);
             this.trStream = trStream;
         }
+
+        @Override
+        public synchronized void print(boolean b) {
+            TrOutputStream.isPrinting.set(true);
+            try {
+                super.print(b);
+            } finally {
+                TrOutputStream.isPrinting.set(false);
+            }
+            super.flush();
```

Comment:
`super.flush()` should be in the finally block too.  Otherwise you could run out of memory adding things to a stream that keeps on throwing an exception.